flink-streaming-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/Kafka09Fetcher.java

```diff
@@ -204,7 +204,22 @@ public void run() {
 			// seek the consumer to the initial offsets
 			for (KafkaTopicPartitionState<TopicPartition> partition : subscribedPartitions()) {
 				if (partition.isOffsetDefined()) {
+					LOG.info("Partition {} has restored initial offsets {} from checkpoint / savepoint; seeking the consumer " +
+						"to position {}", partition.getKafkaPartitionHandle(), partition.getOffset(), partition.getOffset() + 1);
+
 					consumer.seek(partition.getKafkaPartitionHandle(), partition.getOffset() + 1);
+				} else {
+					// for partitions that do not have offsets restored from a checkpoint/savepoint,
+					// we need to define our internal offset state for them using the initial offsets retrieved from Kafka
+					// by the KafkaConsumer, so that they are correctly checkpointed and committed on the next checkpoint
+
+					long fetchedOffset = consumer.position(partition.getKafkaPartitionHandle());
```

Comment:
I just confirmed this with a simple test.
Yes, when no commit offset is available for the group id, the "auto.offset.reset" behavior is used.
The "position" basically means which offset the KafkaConsumer will start reading from. It automatically determines the position once it is assigned partitions, either from committed offsets or the "auto.offset.reset" if none exists.